test/NuGet.Core.Tests/NuGet.Packaging.Test/PackageBuilderTest.cs

```diff
@@ -2934,48 +2986,290 @@ public void PackageBuilderPreserveFileLastWriteTime()
+
+                            // Please note: ZipArchive stream reader sometime changes LastWriteTime by another 1 second off than what "entry.LastWriteTime" has.
+                            // The FAT filesystem of DOS has a timestamp resolution of only two seconds; ZIP file records mimic this.
+                            // As a result, the built -in timestamp resolution of files in a ZIP archive is only two seconds, though extra fields can be used to store more precise timestamps.
+                            // That is why you see this datetime interval instead of actual == of datetimes.
+                            if (File.Exists(path))
+                            {
+                                if (path == before2107File1)
+                                {
+                                    // File from 2020
+                                    Assert.True(entry.LastWriteTime.DateTime >= year2020Date.AddSeconds(-1) && entry.LastWriteTime.DateTime <= year2020Date);
+                                    numberOfDateNotCorrectedFiles++;
+                                }
+                                else if (path == before2107File2 || path == before2107File3)
+                                {
+                                    Assert.True(entry.LastWriteTime.DateTime >= ZipFormatMaxDate.AddSeconds(-2) && entry.LastWriteTime.DateTime <= ZipFormatMaxDate.AddSeconds(-1));
+                                    numberOfDateNotCorrectedFiles++;
+                                }
+                                else if (path == before2107File4)
+                                {
+                                    Assert.True(entry.LastWriteTime.DateTime == ZipFormatMaxDate);
+                                    numberOfDateNotCorrectedFiles++;
+                                }
+                                else
+                                {
+                                    // File from 12/31/2107, 23:59:58 UTC and after that
+                                    Assert.True(entry.LastWriteTime.DateTime >= ZipFormatMaxDate.AddSeconds(-1) && entry.LastWriteTime.DateTime <= ZipFormatMaxDate);
+                                    numberOfDateCorrectedFiles++;
+                                }
+                            }
+                        }
+                    }
+                }
+
+                Assert.True(numberOfDateNotCorrectedFiles == 4);
+                Assert.True(numberOfDateCorrectedFiles == 5);
+                Assert.Equal(innerLogger.LogMessages.Count, 1);
+                string[] logMessages = innerLogger.LogMessages.First().Message.Split('\n');
+                Assert.Equal(logMessages.Count(l => l.Contains("which is out of range of what the zip format supports.")), 5);
             }
         }
 
         [Fact]
-        public void EmitRequireLicenseAcceptance_ShouldThrow()
+        public async Task PackAndExtract_TestLastWriteTimeForZipfileUsingLocalTimeAsync_Succeeds()
```

Comment:
I don't understand what this test, and the next test (about UTC time) are supposed to be testing. They write a file to the filesystem, and then pack a nupkg, where pack reads files off the filesystem. The filesystem normalizes the timezone, so in effect both tests are testing the same thing, and we're not actually testing local time zone vs UTC time zone.